### PR TITLE
Implement SupervisedScheduler::detachThread [BTS-1670]

### PR DESCRIPTION
**Pull request by neunhoef**
This is a backport of https://github.com/arangodb/arangodb/pull/20036

This is one measure to resolve
  https://arangodb.atlassian.net/browse/BTS-1670

In this PR we create a possiblity that a SchedulerThread can notice that 
it might be up to some longer running task (like waiting for a lock) and 
then voluntarily decide to detach itself from the scheduler to make room
for a new thread to avoid starvation of jobs on the queue.

For now, we use this for acquiring a collection level lock (as read lock 
for write operations and as write lock for exclusive access), if we 
cannot acquire the lock within 1s. This will then allow the scheduler
to create new threads to continue work on whatever priority is suitable.

### Scope & Purpose

*(Please describe the changes in this PR for reviewers, motivation, rationale - **mandatory**)*

- [*] :hankey: Bugfix

### Checklist

- [*] Tests
  - [*] C++ **Unit tests**
  - [*] **integration tests**
- [*] :book: CHANGELOG entry made
- [*] Backports
  - [*] Backport for 3.11: This is it.

#### Related Information

*(Please reference tickets / specification / other PRs etc)*

- [*] GitHub issue / Jira ticket: https://arangodb.atlassian.net/browse/BTS-1670
- [*] devel PR: https://github.com/arangodb/arangodb/pull/20036

### CODE REVIEW

File: tests/js/client/server_parameters/detach-threads-cluster.js [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+        for (let i = 0; i < 100; ++i) {
+          let r = arango.POST_RAW(`/_api/document/${cn}`, {_key:"L"+i}, {"x-arango-async": "store", "x-arango-trx-id": trx});
+          assertEqual(202, r.code);
+          assertTrue(r.headers.hasOwnProperty("x-arango-async-id"));
+          l.push(r.headers["x-arango-async-id"]);
+        }
+
+        // Then perform simple reads to the collection. They should soon be
+        // unblocked, because of detached threads.
+        let startTime = new Date();
```
Comment (jsteemann) [inline tests/js/client/server_parameters/detach-threads-cluster.js:138]: `startTime` is not used later.

File: tests/js/client/server_parameters/detach-threads-cluster.js [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+        let guck = c.document("K1");
```
Comment (jsteemann) [inline tests/js/client/server_parameters/detach-threads-cluster.js:139]: `guck` is not used later. So either the assignment to `guck` or the whole statement can be removed.

**Review by jsteemann — APPROVED** [review]
LGTM

**Comment by KVS85** [comment]
- [x] https://jenkins.arangodb.biz/job/arangodb-matrix-pr/32039/

**Pull request merged**